src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -1292,23 +1292,23 @@ public static int ObjToHashAlgId(Object hashAlg)
                 if (hashAlg is SHA512)
                     return CapiHelper.CALG_SHA_512;
             }
-            else if (hashAlg is Type)
-            {
-                TypeInfo hashAlgTypeInfo = ((Type)hashAlg).GetTypeInfo();
 
-                if (typeof(MD5).GetTypeInfo().IsAssignableFrom(hashAlgTypeInfo))
+            Type hashAlgType = hashAlg as Type;
+            if ((object)hashAlgType != null)
```

Comment:
`Type` overloads `==` so casting to `object` means that operator method isn't called.